### PR TITLE
[Console] optimisation getting the command when finding

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | none  
| License       | MIT
| Doc PR        |

Following discussion with @chalasr in here https://github.com/symfony/symfony/pull/31261#discussion_r281033178, first commit will be deleted  when the other PR is merged.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Time for rebase :)

**Comment by Simperfit** [comment]
@nicolas-grekas rebased

File: src/Symfony/Component/Console/Application.php [hunk 1]
```diff
@@ -684,7 +687,7 @@ public function find($name)
             throw new CommandNotFoundException(sprintf("Command \"%s\" is ambiguous.\nDid you mean one of these?\n%s", $name, $suggestions), array_values($commands));
         }
 
-        return $this->get($exact ? $name : reset($commands));
+        return $this->get($name);
```
Comment (chalasr) [inline src/Symfony/Component/Console/Application.php]: `return $this->get(reset($commands));`

**Review by chalasr — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Console/Application.php (hunk 1 above)
Comment (Simperfit) [reply src/Symfony/Component/Console/Application.php]: done

**Comment by Simperfit** [comment]
Status: Needs Review

**Review by chalasr — CHANGES_REQUESTED** [review]
Blocking merge for now, I realize that calling `has()` involves to instantiate some lazy commands which might be a problem even for the related bugfix, I'm going to double-check the behavior tonight.

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @Simperfit.

**Pull request merged**